### ISSUE TITLE
[homework: doing, reading, watching] Linear, logistic regressions and embedding visualizations of Zulip data

### ISSUE THREAD

**Issue by Devadeut** [onefact/datathinking.org#130]
## Doing
- [x] Clean Data Thinking Zulip chat data, located at https://github.com/onefact/datathinking.org-codespace/blob/main/data/datathinking.zulipchat.com/raw/messages-000001.json - put it in a `polars` dataframe and compute summary statistics of the dataset
- [x] Analyze this Zulip chat data using logistic regression, linear regression, and embeddings with the tools we have learned in the lectures (don't forget to ask ChatGPT, Claude, Lex, GPT-4 for help as much as you need, and ask for help on the Data Thinking Zulip chat :)
- [x] Create a visualization of logistic regression of the Data Thinking Zulip chat data
- [x] Create a visualization of linear regression applied to the Data Thinking Zulip chat data 
- [x] Create a visualization of embeddings using the Data Thinking Zulip Chat data
- [x] Make a copy of the Overleaf template: https://www.overleaf.com/read/ghpyzqwqwxpv (need to create an account and/or sign in if this is your first time using Overleaf). To make a copy, open the project after signing in using this link, and click on `Menu`, then `Copy Project`:
<img width="310" alt="image" src="https://user-images.githubusercontent.com/5317244/226241154-a96f364e-6892-4229-b363-3a507b5dc2f2.png">

- [x] In Overleaf, edit the template and figure out how to include a PDF figure in the report, alongside a brief description (a few sentences or paragraphs is fine!) of each of the analyses you performed, why you chose them, and the math equation for the linear regression, logistic regression, and embedding you used.
- [x] Add the PDF of the report to this issue as a comment.
- [x] Send a message on Zulip with a link to this comment, alongside the image representing your favorite visualization

## Reviewing
- [x] Review how Jaan got unstuck in the lecture recordings at https://panopto.ut.ee/Panopto/Pages/Sessions/List.aspx?folderID=43bb180c-79a6-4324-b055-afa400ecd1a0
- [ ] Review collaborative whiteboards from past classes: listed at https://www.datathinking.org/university-of-tartu
- [x] Review Jupyter notebooks from past classes: 
    - https://github.com/onefact/datathinking.org-codespace/blob/main/notebooks/230313-embeddings-volume-2.ipynb
    - https://github.com/onefact/datathinking.org-codespace/blob/main/notebooks/230316-embeddings-text-data-like-chat-logs-and-networks.ipynb
    - https://github.com/onefact/datathinking.org-codespace/tree/main/notebooks

## Reading
- [ ] [inspiration for how to categorize discussion data in `json` format with chatgpt] https://genmon.github.io/braggoscope/about & https://news.ycombinator.com/item?id=35073603
- [x] [help with helping you learn to be a prompt engineer] https://github.com/dair-ai/Prompt-Engineering-Guide
- [x] [help with storytelling] https://themarkup.org/hello-world/2023/02/04/journalistic-lessons-for-the-algorithmic-age
- [x] [help with debugging] https://wizardzines.com/images/debugging/toc-letter.pdf
- [ ] [design, visual communication] https://anthonyhobday.com/sideprojects/saferules/ & https://news.ycombinator.com/item?id=34684761
- [x] [background on ChatGPT and Claude from an ex-OpenAI, Anthropic founder] read `The Boy Whose Light Went Out` by Jack Clark http://techpolicylab.uw.edu/wp-content/uploads/2022/04/Telling_Stories_Pages_4-4-22.pdf 
- [ ] [history of OpenAI internal difficulties] https://www.technologyreview.com/2020/02/17/844721/ai-openai-moonshot-elon-musk-sam-altman-greg-brockman-messy-secretive-reality/
- [x] [context for supply chain ban on GPU exports/imports] https://www.palladiummag.com/2021/10/11/the-triumph-and-terror-of-wang-huning/
- [x] [starting to think about bias in what you read from chatgpt or elsewhere] https://catalogofbias.org/biases/
- [x] [psychological tools for support in dealing with ChatGPT, especially if you decide to try jailbreaks] https://www.bellingcat.com/resources/2022/11/23/how-to-maintain-mental-hygiene-as-an-open-source-researcher/
- [ ] [how data is being used by open source intelligence researchers in war] https://www.cbsnews.com/news/bellingcat-russia-putin-ukraine-60-minutes-2022-08-21/
- [ ] [background on social media and recommendation algorithms] https://knightcolumbia.org/content/understanding-social-media-recommendation-algorithms

## Watching
(message Jaan if you need a VPN or these links don't work)
- [x] [context for popular media portrayals of chatgpt] South Park S26E04 - "Deep Learning" - https://southpark.cc.com/episodes/8byci4/south-park-deep-learning-season-26-ep-4 
- [x] [context for how the CEO of OpenAI talks about it] https://abcnews.go.com/Technology/video/openai-ceo-cto-risks-ai-reshape-society-97949497 [article](https://abcnews.go.com/Technology/openai-ceo-sam-altman-ai-reshape-society-acknowledges/story?id=97897122)
- [x] [an emotional history of the world; context for our data thinking practice of feel, ask, do, think -- and how the thoughts in people's heads can shape society] 1 episode of https://thoughtmaybe.com/cant-get-you-out-of-my-head/

**Comment by Devadeut** [comment]
[Report.pdf](https://github.com/onefact/datathinking.org/files/11414216/Report.pdf)

**Comment by nshut** [comment]
**Hey,

here is my review of your report;**

I don’t have time to structure my critique, but I will first focus on negative points and finish with the positive. The title is ‘Logistic Regression‘, which gives the impression that it is the main focus of your work, but this is just one out of three accomplishments. Therefore, the title might be a bit misleading. Your position, ‘Sophomore CSE‘, is centered but should be below your name. I think this is a problem with LaTeX formatting. I am also not sure how relevant the term ‘sophomore‘ is for this part. In the introduction, you claim that you will explain 'why I chose them,' referring to the analyses you used, but you didn’t do it in the text. There is inconsistent formatting in the 'Embeddings' part of your report. It seems that you forgot to edit the title that is displayed on non-title pages - your report just uses ‘Running Title‘ and ‘Lastname‘ for this purpose. The legend of the second plot is, for some reason, in the middle of the plot.

What I like about your work: all parts are of the same size. The report is, let's say, laconic and easy to read and navigate. Also, the first two plots are nice. Linear regression actually represents that blue and red points are different. All figures are numbered. The last one is three-dimensional. It looks cool, and different shades of blue are used.

**and here is ChatGPT's:**

1. Writing:

The writing is generally clear, but it could be more detailed. For example, in the Introduction, you mention that you will describe each of the analyses you performed, but the descriptions in the following sections are quite brief.

There are a few grammatical errors and awkward phrases, such as "Here I will briefly describe each of the analyses I performed, why I chose them, and the math equation for the linear regression, logistic regression, and embedding I used." This sentence could be rephrased for clarity, perhaps something like: "In this report, I will provide a brief description of each analysis I performed, explain why I chose these particular methods, and present the mathematical equations used in the linear regression, logistic regression, and embedding techniques."

The conclusion is very brief and could be expanded to provide a more detailed summary of what you learned and the results of your analyses.

2. Visualization:

Without seeing the actual visualizations, it's hard to critique them. However, based on your descriptions, it seems like you've used appropriate methods for visualizing your data.

Make sure your visualizations are clearly labeled and include a legend where necessary. This will make it easier for readers to understand what they're looking at.

3. Content:

The content is generally good, but it could be more detailed. For example, in the section on embeddings, you mention that you're not sure how you did the embedding. It would be beneficial to understand this process more fully and explain it in more detail in your report.

The report would benefit from a more detailed discussion of the results of your analyses. For example, how well did the linear and logistic regression models perform? What insights were you able to gain from the embeddings?

4. Improvements:

The report could benefit from more detailed explanations of your methods and results. This would make it more informative and valuable to readers.

Including the actual visualizations in the report would also be a significant improvement. This would allow readers to see the results of your analyses for themselves.

Finally, proofreading the report carefully before submission would help catch grammatical errors and improve the overall quality of the writing.

I hope these critiques are helpful! Let me know if you have any questions or if there's anything else you'd like me to review.
